doc/en/source/ngsildbasediotintegration.rst

```diff
@@ -0,0 +1,476 @@
+    
+     git clone https://github.com/smartfog/fogflow.git
+     cd fogflow/test/iot-agent/
+
+    ./services orion
+  
+     If you want to clean up and start over again you can do this with the following command:
+    
+    ./services stop
+  
+
+Before you start the following steps, please check if your Orion-LD broker and FogFlow system is running properly.
+
+# check if the orion-ld broker is running
+
+.. code-block:: console
+
+	curl localhost:1026/ngsi-ld/ex/v1/version
+
+The response will look similar to the following:
+
+.. code-block:: console
+
+	{
+  		"Orion-LD version": "0.7.0",
+  		"based on orion": "1.15.0-next",
+  		"kbase version": "0.6",
+  		"kalloc version": "0.6",
+  		"khash version": "0.6",
+  		"kjson version": "0.6",
+		"boost version": "1_62",
+		"microhttpd version": "0.9.72-0",
+		"openssl version": "OpenSSL 1.1.0l  10 Sep 2019",
+		"mongo version": "1.1.3",
+		"rapidjson version": "1.0.2",
+		"libcurl version": "7.52.1",
+		"libuuid version": "UNKNOWN",
+		"branch": "",
+		"cached subscriptions": 0,
+		"Next File Descriptor": 19
+	}
+
+# check if the IoT Agent broker is running
+
+.. code-block:: console
+
+	curl localhost:4041/iot/about
+
+The response will look similar to the following:
+
+.. code-block:: console
+
+	{
+		"libVersion": "2.15.0",
+		"port": "4041",
+		"baseRoot": "/",
+		"version": "1.16.0"	
+	}
+    
+# check if the IoT Device (Sensor, Actuator)is running properly
+	
+	Check the status from the Device dashboard
+
+	You can open the device dashboard in your web browser to see the current system status via the URL: **localhost:3000/device/monitor**
+	
+	Once you are able to access the device dashboard, you can see the following web page
+
+.. figure:: figures/device.png
+
+# check if the FogFlow system is running properly
+	
+	Check the system status from the FogFlow DashBoard
+
+	You can open the FogFlow dashboard in your web browser to see the current system status via the URL: **localhost:80**
+	
+	Once you are able to access the FogFlow dashboard, you can see the following web page
+
+.. figure:: figures/dashboard.png
+
+
+
+How to Fetch data from IoT Agent to Orion-LD
+================================================================
+
+**Step1** Provisioning a Service Group
+-----------------------------------------------------------------
+Invoking group provision is always the first step in connecting devices since it is always necessary to supply an authentication key with each measurement and the IoT Agent will not initially know which URL the context broker is responding on.
+
+It is also possible to set up default commands and attributes for all anonymous devices as well, but this is not done within this tutorial as we will be provisioning each device separately.
+
+This example provisions an anonymous group of devices. It tells the IoT Agent that a series of devices will be sending messages to the IOTA_HTTP_PORT (where the IoT Agent is listening for Northbound communications)
```

Comment:
after northbound communication we can also write "- data coming from  the IOT device"

Reply:
Done